agent/connect/ca/testing.go

```diff
@@ -263,3 +263,8 @@ func ApplyCARequestToStore(store *state.Store, req *structs.CARequest) (interfac
 		return nil, fmt.Errorf("Invalid CA operation '%s'", req.Op)
 	}
 }
+func requireTrailingNewline(t testing.T, leafPEM string) {
+	if '\n' != rune(leafPEM[len(leafPEM)-1]) {
```

Comment:
minor: I guess we should check that `len(leafPEM) > 0` before taking the `-1` index to prevent a panic when a test has an unexpected result.